### PR TITLE
SOLR-11262: XML writer implements writeMap and writeIterator

### PR DESCRIPTION
**Pull request by munendrasn**

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/response/TestPushWriter.java [hunk 1]
```diff
@@ -45,26 +46,47 @@
...
+      m = (Map) Utils.fromJavabin(baos.toByteArray());
     }
     checkValues(m);
   }
 
+  public void testXmlWriter() throws Exception {
+    try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        OutputStreamWriter osw = new OutputStreamWriter(baos, StandardCharsets.UTF_8)) {
+      XMLWriter xml = new XMLWriter(osw,
+          new LocalSolrQueryRequest(null, new ModifiableSolrParams()), new SolrQueryResponse());
```
Comment (noblepaul) [inline solr/core/src/test/org/apache/solr/response/TestPushWriter.java:73]: what is this `LocalSolrQueryRequest` for?

Comment (munendrasn) [reply solr/core/src/test/org/apache/solr/response/TestPushWriter.java:73]: XmlWriter takes SolrQueryRequest as one of the arg. so, I have created localSolrQueryRequest

**Comment by noblepaul** [comment]
Add an entry to the CHANGES.txt as well and I can merge this straight away

**Comment by munendrasn** [comment]
Thanks @noblepaul

**Pull request merged**